napari/layers/utils/_tests/test_layer_utils.py

```diff
@@ -238,3 +239,25 @@ def test_coerce_current_properties_invalid_values():
 
     with pytest.raises(ValueError):
         _ = coerce_current_properties(current_properties)
+
+
+@pytest.mark.parametrize(
+    "dims_displayed,ndim_world,ndim_layer,expected",
+    [
+        ([1, 2, 3], 4, 4, [1, 2, 3]),
+        ([0, 1, 2], 4, 4, [0, 1, 2]),
+        ([1, 2, 3], 4, 3, [0, 1, 2]),
+        ([0, 1, 2], 4, 3, [2, 0, 1]),
```

Comment:
@kevinyamauchi : sorry to dig this PR up, but I'm working on #4926, almost have a fix (that uses the function being tested), but this expected result blocks me.

I've read through #3848 and #3882, but still fail to understand the expected result here. Can you help explain it? Is it a desired result, or just the behavior of napari at the moment?

Reply:
Hey @andy-sweet ! No worries. I understand this is confusing - probably should have left a note.  This is just the behavior as napari was at the time. Basically, it is just checking to make sure that the behavior matches the legacy code below that set the "rule" for matching layer dimensions with displayed dimensions and broadcasting lower dimensionality layer coordinate spaces into higher dimensional world coordinate spaces. It's from super early on in napari (I can't remember if there was even any explicit discussion about it). As you probably picked up from reading #3882, there is interest in changing how the broadcasting works, but we didn't come up with a better solution. I'll head over to #4926 and see if you've come up with a more intuitive scheme 😄 

https://github.com/napari/napari/blob/74fd219b59c64b58296a9ecc13a96776f255d0bc/napari/layers/base/base.py#L961-L967

Reply:
Thanks for the info. I don't have a solution for this in #5004 to fix #4926, but there could be one. It sounds like we might be OK making that change as this test case just reflects existing behavior that may not be desired.